### DISCUSSION TITLE
plot_data_plane error

### DISCUSSION THREAD

**Discussion by jason-english** [dtcenter/METplus#1336] [Plot Generation] [answered]
Does anyone know how to resolve this error with plot_data_plane? It happens both on Jet and Hera, and on two different grib2 files I've tried (One was HRRRv4 output, and one was RRFS Dev 1 output) (two different precip fields).  Thanks.

hfe09:/scratch2/BMC/zrtrr/jenglish/met/METplus-4.0.0_Tutorial>plot_data_plane wrfnat_hrconus_12.grib2 wrfnat_hrconus_12.ps APCP_P8_L1_GLC0_acc12h
ERROR  :
ERROR  : yyerror() -> syntax error in file "/tmp/met_config_253890_0"
ERROR  :
ERROR  :    line   = 2
ERROR  :
ERROR  :    column = 22
ERROR  :
ERROR  :    text   = "(nul)"
ERROR  :
ERROR  :
ERROR  : (nul)
ERROR  : (nul)
ERROR  :

**Comment by georgemccabe** [comment]
Hi @jason-english, the error you are receiving is due to the field information being improperly formatted. The field information must contain the name and level attributes.

There are some examples towards the bottom of this section of the MET User's Guide:
https://met.readthedocs.io/en/latest/Users_Guide/plotting.html?highlight=Plot-Data-Plane#plot-data-plane-usage

It looks like you want to read the 12 hour accumulation of the APCP field. Here is an example of how to do that:

` plot_data_plane wrfnat_hrconus_12.grib2 wrfnat_hrconus_12.ps 'name="APCP"; level="A12";'`

There are many additional options that can be provided besides name and level to extract the correct data from a GRIB file. To see a list of the possible options, navigate to:
https://met.readthedocs.io/en/latest/Users_Guide/config_options.html#settings-common-to-multiple-tools
and search for **File-format specific settings for the “field” entry**.

**Reply by jprestop** [reply]
@jason-english Just to add to @georgemccabe's answer, there is also a page for the METplus Tutorial that goes over formatting for the field string:
https://dtcenter.org/metplus-practical-session-guide-version-4-0/session-1-metplus-setupgrid-grid/met-tool-plot-data-plane/field-string

**Reply by jason-english** [reply]
Hmm, so I tried the corrected syntax for my grib2 file and now I have this error: 

jetfe5:/lfs1/BMC/wrfruc/jenglish/metplus/METplus-4.0.0_Tutorial>plot_data_plane wrftwo_rr_06.grib2 wrftwo_rr_06_pcp.ps 'name="APCP"; level="A12";'
DEBUG 1: Opening data file: wrftwo_rr_06.grib2
ERROR  : 
ERROR  : MetGrib2DataFile::read_grib2_grid() -> found unrecognized grid definition (32769)
ERROR  :

**Comment by JohnHalleyGotway** [comment, marked as answer]
Hi @jason-english I grabbed your sample file from jet, tried running it through MET, and see the same behavior you did:
```
ERROR : MetGrib2DataFile::read_grib2_grid() -> found unrecognized grid definition (32769)
```
Running it through wgrib2 (with -V) produces this output:
```
51:24053677:vt=2019020514:surface:0-6 hour acc fcst:APCP Total Precipitation [kg/m^2]:
    ndata=794802:undef=0:mean=1.04395:min=0:max=93.4375
    grid_template=32769:winds(grid):
	I am not an Arakawa E-grid. 
	I am rotated but have no rotation angle.
	I am staggered. What am I?
	(953 x 834) units 1e-06 input WE:SN output WE:SN res 56
	lat0 -10.590603 lat-center 54.000000 dlat 121.813000
	lon0 220.914154 lon-center 254.000000 dlon 121.813000 #points=794802
```
Separately, I ran it through NASA's Panoply tool and am able to plot it:
<img width="1052" alt="Screen Shot 2022-01-06 at 9 59 50 AM" src="https://user-images.githubusercontent.com/21087144/148421841-7ea3b8d2-1538-4734-91e7-31ef2ba398ab.png">

I believe that WRF's ability to write GRIB2 output directly is relatively new. Unfortunately, this GRIB2 file is not on a grid that's currently supported by MET:
   https://met.readthedocs.io/en/develop/Users_Guide/appendixB.html?highlight=lambert#map-projections

There is a proposal underway to secure funding to enhance MET to better integrate with WRF's recent enhancements. But supporting this grid will require code changes. For MET version 10.0.0, your options are:
- Post-process WRF NetCDF output files using the Unified PostProcessor (UPP) which writes GRIB2 files and interpolates to one of the supported projections.
- Use some other GRIB2 regridding tool to interpolate to a supported projection type.
- Write a python script to do that regridding in memory and serve the output up to the MET tools via python embedding.

**Reply by jason-english** [reply]
This is surprising to me -- this is HRRR standard output on Lambert Conformal Projection, which is WRF standard. Is it because the HRRR domain is a subset of the globe...?

**Reply by JohnHalleyGotway** [reply]
@jason-english, OK, interesting. I went to [nomads](https://nomads.ncep.noaa.gov/pub/data/nccf/com/hrrr/prod/hrrr.20220111/conus/) to pull another GRIB2 file. The output from wgrib2 looks much more sane:

```
wget https://nomads.ncep.noaa.gov/pub/data/nccf/com/hrrr/prod/hrrr.20220111/conus/hrrr.t01z.wrfsfcf06.grib2
wgrib2 -d 90 -V hrrr.t01z.wrfsfcf06.grib2
90:64274546:vt=2022011107:surface:5-6 hour acc fcst:APCP Total Precipitation [kg/m^2]:
    ndata=1905141:undef=0:mean=0.0553217:min=0:max=75.88
    grid_template=30:winds(grid):
	Lambert Conformal: (1799 x 1059) input WE:SN output WE:SN res 8
	Lat1 21.138123 Lon1 237.280472 LoV 262.500000
	LatD 38.500000 Latin1 38.500000 Latin2 38.500000
	LatSP 0.000000 LonSP 0.000000
	North Pole (1799 x 1059) Dx 3000.000000 m Dy 3000.000000 m 
```

And MET is able to plot this data just fine:

```
plot_data_plane hrrr.t01z.wrfsfcf06.grib2 apcp.ps 'name="APCP"; level="A1";'
```

<img width="573" alt="Screen Shot 2022-01-11 at 10 20 48 AM" src="https://user-images.githubusercontent.com/21087144/148990738-7fcf4a2c-aed2-463d-83fe-69c128552743.png">

How does this GRIB2 file differ from the GRIB2 files you're using?

**Reply by JohnHalleyGotway** [reply]
@jason-english just wanted to follow up on this discussion. Looks like it's been 2 weeks since we last talked. Wondering if you've made any progress diagnosing this behavior? How does the structure of your HRRR GRIB2 file compare to what's available on NOMADS?

**Reply by JohnHalleyGotway** [reply]
I'll go ahead and mark this discussion as being answered and "Lock" it. If more issues or questions arise, please feel free to start a new discussion.
